### PR TITLE
Refactor caching strategy for FASTA search

### PR DESCRIPTION
**Pull request by namvdo**

### CODE REVIEW

File: complearn-genbank/app.ts [hunk 1]
```diff
@@ -4,16 +4,16 @@ import cors from "cors";
 import session from "express-session";
 import {Strategy as GoogleStrategy, Profile as GoogleProfile} from "passport-google-oauth20";
 import {Strategy as GitHubStrategy, Profile as GithubProfile} from "passport-github2";
-import ENV_LOADER from "./configurations/envLoader";
-import logger from "./configurations/logger";
+import ENV_LOADER from "./configurations/envLoader.js";
```
Comment (rudi-cilibrasi) [inline complearn-genbank/app.ts:7]: Shawn better check to see if this .js import is correct

File: complearn-genbank/routes/redis.ts [hunk 2]
```diff
@@ -22,55 +22,126 @@ connectToRedis();
...
+            }
+        });
     }
-});
+);
 
+router.post(
+    "/incr",
+    (req: Request<{}, any, RedisIncrRequest>, res: Response) => {
+        return new Promise<void>(async (resolve) => {
```
Comment (rudi-cilibrasi) [inline complearn-genbank/routes/redis.ts:104]: should not use new Promise
should declare this anonymous function `async` instead

File: complearn-genbank/routes/redis.ts [hunk 3]
```diff
@@ -22,55 +22,126 @@ connectToRedis();
...
+            }
+        });
     }
-});
+);
 
+router.post(
+    "/del",
+    (req: Request<{}, any, RedisDelRequest>, res: Response) => {
+        return new Promise<void>(async (resolve) => {
```
Comment (rudi-cilibrasi) [inline complearn-genbank/routes/redis.ts:127]: do not need new Promise here

File: ncd-calculator/src/services/CompressionService.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            return new Worker(URL.createObjectURL(blob));
+        }
+    }
+
+    static async processFiles(
+        contents: string[],
+        labels: string[],
+        onProgress?: (i: number, j: number, value: number) => void
+    ): Promise<number[][]> {
+        return new Promise((resolve, reject) => {
```
Comment (rudi-cilibrasi) [inline ncd-calculator/src/services/CompressionService.ts:43]: do not use Promise(resolve, reject) form.
use async function form
and throw errors (this becomes a reject in an async function)
or return success. never use resolve/reject explicitly when you can avoid it using async/await

**Review by rudi-cilibrasi — APPROVED** [review]

File: complearn-genbank/Dockerfile [hunk 5]
```diff
@@ -1,17 +1,27 @@
...
 WORKDIR /app
 
-# Copy package files
 COPY package*.json ./
 
-# Install dependencies
 RUN npm install
 
-# Copy source code
 COPY . .
```
Comment (Sonnpm197) [inline complearn-genbank/Dockerfile:10]: why do you remove this?
the dist has to be removed everytime we build docker image and rebuild again using npx tsc

**Pull request merged**